tests/comfy_cli/test_knowledge.py

```
@@ -0,0 +1,622 @@
+        assert all(len(v) == len(set(v)) for v in bundle.templates.values())
+
+    def test_deprecations_keyed_by_id(self, bundle):
+        assert bundle.deprecations["kling-avatar-2"]["superseded_by"] == "sync-3"
+        assert set(bundle.deprecations) == {"kling-avatar-2", "sora-2"}
+
+    def test_resolve(self, bundle):
+        assert knowledge.resolve(bundle, "  HAILUO 03 ")["id"] == "minimax-h3"
+        assert knowledge.resolve(bundle, "Kling-Avatar-2")["id"] == "kling-avatar-2"
+        assert knowledge.resolve(bundle, "nope-xyz") is None
+
+    def test_resolve_accepts_spelling_variants(self, bundle):
```

Pylint (4.0.6)

[convention] Missing function or method docstring

(C0116)